Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/dialog/DialogFragmentTemplate.kt

```diff
@@ -0,0 +1,82 @@
+package de.rki.coronawarnapp.ui.dialog
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.appcompat.app.AlertDialog
+import androidx.core.os.bundleOf
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.setFragmentResult
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.util.ContextExtensions.getColorCompat
+import kotlinx.parcelize.Parcelize
+import timber.log.Timber
+
+class DialogFragmentTemplate : DialogFragment() {
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val config = requireArguments().getParcelable<Config>(PARAM_DIALOG_CONFIG)
+        requireNotNull(config) { "DialogFragment config is null" }
+        isCancelable = config.isCancelable
+
+        return config.run {
+            MaterialAlertDialogBuilder(requireContext())
+                .setTitle(titleRes)
+                .setMessage(messageRes)
+                .setPositiveButton(positiveButtonRes) { _, _ -> setAction(Action.PositiveButtonClicked) }
+                .setNegativeButton(negativeButtonRes) { _, _ -> setAction(Action.NegativeButtonClicked) }
+                .setNeutralButton(neutralButtonRes) { _, _ -> setAction(Action.NeutralButtonClicked) }
+                .create()
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        val config = requireArguments().getParcelable<Config>(PARAM_DIALOG_CONFIG)
+        if (config?.isDeleteDialog == true) {
+            (dialog as AlertDialog).getButton(DialogInterface.BUTTON_POSITIVE)
+                ?.setTextColor(requireContext().getColorCompat(R.color.colorTextDeleteButtonDialog))
+        }
+    }
+
+    private fun setAction(action: Action) {
+        Timber.d("DialogFragment setAction(action=%s", action)
+        setFragmentResult(REQUEST_KEY, bundleOf(PARAM_DIALOG_ACTION to action))
+    }
+
+    override fun onDismiss(dialog: DialogInterface) {
+        Timber.d("DialogFragment dismissed")
+        setAction(Action.Dismissed)
+        super.onDismiss(dialog)
+    }
+
+    enum class Action {
+        PositiveButtonClicked,
+        NegativeButtonClicked,
+        NeutralButtonClicked,
+        Dismissed
+    }
+
+    @Parcelize
+    data class Config(
+        val titleRes: String,
+        val messageRes: String,
+        val positiveButtonRes: String,
+        val negativeButtonRes: String?,
+        val neutralButtonRes: String?,
+        val isDeleteDialog: Boolean = false,
```

Comment:
This seems to be used to set the color of the positive button to red. I would prefer a more obvious and direct way, eg. color of positive button, so I can use orange or color a dialog button although it is not for deletion.

Reply:
I'm not sure if I'd change this one. I agree that it'd be nice to use it like you described if we would truly have different colors for dialog buttons, but since it's only red, I'd keep it like this. We didn't have any differently colored dialog buttons up until recently #5538 and other than the red negative buttons, I doubt we'll have other different colors in the future. I'd prefer to just pass a `true` when creating a dialog button than having to pass the res id of the red color with the hope we'll need to use different colors in the future which is highly unlikely to change.